File: output/qwen_2.5_72b_instruct/avatar/direct_translation/java/python/failed_translation/atcoder_ABC125_C.py

import math

class atcoder_ABC125_C:
    @staticmethod
    def main():
        sc = input().split()
        N = int(sc[0])
        A = list(map(int, sc[1:]))
        M = [0] * N
        L = [0] * N
        R = [0] * N
        L[0] = A[0]
        for i in range(1, N):
            L[i] = math.gcd(L[i - 1], A[i])
        R[N - 1] = A[N - 1]
        for i in range(N - 2, -1, -1):
            R[i] = math.gcd(R[i + 1], A[i])
        answer = max(L[N - 2], R[1])
        for i in range(1, N - 1):
            answer = max(answer, math.gcd(L[i - 1], R[i + 1]))
        print(answer)

if __name__ == "__main__":
    atcoder_ABC125_C.main()

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/failed_translation/atcoder_ABC125_C.java

import java.util.Scanner;

class Main {
    public static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    public static int[] ruiseki_lr(int[] array) {
        java.util.function.BiFunction<Integer, Integer, Integer> op = (a, b) -> gcd(a, b);
        int e = 0;
        int n = array.length;
        int[] left = new int[n + 1];
        int[] right = new int[n + 1];
        for (int i = 0; i < n; i++) {
            left[i + 1] = op.apply(left[i], array[i]);
        }
        for (int i = n - 1; i >= 0; i--) {
            right[i] = op.apply(right[i + 1], array[i]);
        }
        return new int[]{left[n], right[0]};
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }
        int[] result = ruiseki_lr(a);
        int ans = 0;
        for (int i = 0; i < n; i++) {
            ans = Math.max(gcd(result[0], result[1]), ans);
        }
        System.out.println(ans);
    }
}

